text/002x-succinct-atomic-swaps.md

```diff
@@ -0,0 +1,139 @@
+
+- Title: succinct-atomic-swaps
+- Author: [Gene Ferneau](mailto:gene@ferneau.link)
+- Start date: March 21, 2021
+- RFC PR: [tbd]()
+- Tracking Issue: [tbd]()
+
+---
+
+# Summary
+[summary]: #summary
+
+Succinct atomic swaps [\[1\]](#references) are a technique developed by Ruben Somsen to atomically exchange funds between mutually untrusting parties using the Bitcoin cryptocurrency.
+The techniques can be adapted to work for the Grin cryptocurrency following guides developed by Jasper van der Maarel [\[2\]](#references). A previous version of the protocol was implemented
+using Hash-based Time Locked Contracts in the [grinswap](https://github.com/GrinSwap/proof-of-concept) wallet project.
+
+# Motivation
+[motivation]: #motivation
+
+This RFC aims to show a succinct atomic swap protocol using adaptor signatures, accompanied by an implementation in the reference Mimblewimble wallet [grin-wallet](https://github.com/geneferneau/grin-wallet/tree/atomic). All Grin wallet implementations should be able to use this RFC to interoperate, and perform atomic swaps with each other.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+In an atomic swap, two parties agree to lock funds on one chain (e.g. Bitcoin), and interactively unlock funds on another chain (e.g. Grin). By unlocking funds on the Grin chain, the secret to unlock the funds on the Bitcoin chain is revealed. Revealing the secret this way allows the exchange to happen "atomically": when one lock is opened, so is the other lock. This atomic structure keeps one party from being able to steal funds from the other party: either both locks are opened, or neither are opened.
+
+The reference implementation uses Bitcoin as the other chain in the swap, but potentially any other chain using Secp256k1 keys and multisignatures can work with the protocol.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Funds on the non-Grin chain need to be locked in a multisignature transaction that requires knowledge of both "atomic nonces" to be spent. A refund transaction is also needed to recover funds in case one of the parties becomes non-responsive, or the protocol needs to be aborted for another reason.
+
+## Atomic nonces
+[atomic-nonces]: #atomic-nonces
+
+In the case of Grin, atomic nonces are just Secp256k1 private keys. Let's say Alice knows atomic nonce `aN`, and Bob knows atomic nonce `bN`.
```

Comment:
You talk about atomic nonces as if you already introduced them. I assume they are the "secret"s to unlock funds you mention above?! If so, that is worth mentioning.

Reply:
Yeah, tried to clarify in the next couple sentences how they are used. Added a snip to the end of the first sentence.